authentication/tests/test_permissions.py

```diff
@@ -0,0 +1,53 @@
+from django.test import TestCase
+from django.contrib.auth.models import User, Group
+from emis.permissions import EmisPermission, EmisPermMode, decompose_perm_code, get_code
+from emis.groups import PERMISSIONS_BY_GROUP
+
+
+class GroupPermissionTests(TestCase):
```

Comment:
I'm thinking for this file - i'd like to see a test that better mirrors the exactly. code flow that the permissions are being set for the user. It could be as simple as moving these 2 methods - create ser in group and check permission into the class that will be used by the views to create users. As per my next comment I think we are just testing these methods work as expect but these 2 static methods are just part of the test class never to be used by the actual application.

Reply:
Added test to create a teacher through authentication:create-user and verify some expected + unexpected permissions.